### PR TITLE
EOS-18604: Uniform distribution of CAS FOM locality index

### PR DESCRIPTION
**Pull request by maximalezhin**
Problem: CAS FOM locality index is calculated as a hash of
catalogue FID.

During performance profiling it has been found that the same
locality index is assigned to FOM if the CAS FOPs contain
the same FID. For instance, typical S3 client performs
operations on a single S3 bucket, which lead to limited set of
indecies (meta-index and ordinary index related to S3 bucket).
In that case, assigned localities' runq and waitq queues get
overloaded, while other localities' queues are not populated
by incoming CAS FOPs. It increases CAS FOP processing latency.

Solution: Locality index is implemented using incremental
static counter. It's updated by +1 value each time cas_fom_home_locality()
function is called.

### CODE REVIEW

**Review by huanghua78 — APPROVED** [review]

File: fop/fom_generic.c [hunk 1]
```diff
@@ -68,7 +68,7 @@ M0_INTERNAL void m0_fom_mod_rep_fill(struct m0_fop_mod_rep *rep,
 				     struct m0_fom *fom)
 {
 	rep->fmr_remid.tri_txid = m0_fom_tx(fom)->t_id;
-	rep->fmr_remid.tri_locality = fom->fo_ops->fo_home_locality(fom);
+	rep->fmr_remid.tri_locality = fom->fo_loc_idx;
```
Comment (madhavemuri) [inline fop/fom_generic.c:71]: Why this change is needed ?

Comment (maximalezhin) [reply fop/fom_generic.c:71]: Yes, the idea of m0_fom_mod_rep_fill() function is to return transaction id and locality index of FOM processing a transaction. This data may be used later for FSYNC request.
When CAS FOM locality assignment is implemented via incremental counter, .fo_home_locality() returns not the same, but new value of locality index. This new value is incorrect in terms that transaction is being processed in another locality, and server returns new value. W/o this fix, FSYNC causes crash (see EOS-18604):

`740967 14037149.844.544072 7efe9b7fdb70 DEBUG fop/fom.c:1693:m0_fom_phase_set fom=0x7efb08070a20, item 0x7efb080704f0[123] phase set: tx_
{{ wait -> start}}
{{ 740968 14037149.844.546078 7efe9b7fdbc0 CALL fop/fom_generic.c:804:m0_fom_tick_generic < fom=0x7efb08070a20 phase=start rc=2}}
{{ 740969 14037149.844.548766 7efe9b7fdbc0 CALL fop/fom_generic.c:805:m0_fom_tick_generic < rc=2}}
{{ 740970 14037149.844.549014 7efe9b7fdc30 CALL mdservice/fsync_foms.c:185:fsync_fom_tick >}}
{{ 740971 14037149.844.549096 7efe9b7fdc30 DEBUG mdservice/fsync_foms.c:198:fsync_fom_tick Target tx: 9}}
{{ 740972 14037149.844.555016 7efe9b7fdbb0 CALL be/engine.c:917:m0_be_engine__tx_find > en=0x7ffea2cc3d80 id=9}}
{{ 740973 14037149.844.560170 7efe9b7fdb70 CALL be/tx.c:376:m0_be_tx_get > tx=0x7efb0806ec28 t_ref=1 state=placed}}
{{ 740974 14037149.844.585398 7efe9b7fda40 FATAL lib/assert.c:50:m0_panic panic: (be_tx_is_locked(tx)) at m0_be_tx_get() (be/tx.c:377}}
{{ ) [git: sage-base-1.0-217-g40760892] /var/log/seagate/motr/trace/m0d-0x7200000000000001:0xc/m0trace.250375}}`

**Review by madhavemuri — APPROVED** [review]

**Comment by huanghua78** [comment]
@maximalezhin please fix the DCO.

**Comment by maximalezhin** [comment]
Signed and fixed the DCO
